content/hardware/06.nicla/boards/nicla-voice/tutorials/user-manual/content.md

```diff
@@ -716,6 +716,193 @@ Upload the example sketch again and open the IDE's Serial Plotter by navigating
+  x_mag_raw = (0x0000 | sensor_data[0] >> 3 | sensor_data[1] << 5);
+  y_mag_raw = (0x0000 | sensor_data[2] >> 3 | sensor_data[3] << 5);
+  z_mag_raw = (0x0000 | sensor_data[4] >> 1 | sensor_data[5] << 7);
+  hall_raw  = (0x0000 | sensor_data[6] >> 2 | sensor_data[7] << 6);
+
+  // Print raw magnetometer data.
+  Serial.print("x_mag_raw:");
+  Serial.print(x_mag_raw);
+  Serial.print(",");
+  Serial.print("y_mag_raw:");
+  Serial.print(y_mag_raw);
+  Serial.print(",");
+  Serial.print("z_mag_raw:");
+  Serial.print(z_mag_raw);
+  Serial.print(",");
+  Serial.print("hall_raw:");
+  Serial.println(hall_raw);
+
+  delay(1000);
+}
+```
+
+Here you can find a step-by-step explanation of the code:
+
+First, the necessary libraries are included: 
+
+- `NDP.h` for the Nicla Voice board's basic functions and the magnetometer control.
+- Macros are defined for checking the status of the magnetometer; these macros allow the sketch to detect and handle sensor errors. 
+
+Next, user functions `ledBlueOn()`, `ledGreenOn()`, and `ledRedBlink()` definition: 
+
+- These functions allow the onboard LEDs to flash specific colors to indicate different states: blue for a successful match, green for an event, and red to indicate an error.
+
+Next, in the `setup()` function:
+
+- The serial communication is initialized at a baud rate of 115200.
+- The Nicla Voice board is initialized, and the LDO regulator (used for putting the board into power-saving mode) is disabled to avoid communication problems with the magnetometer. 
+- Error and event handlers are initialized.
+- NDP processor is initialized; this process includes populating the external Flash memory of the board with the NDP processor's internal microcontroller firmware (`mcu_fw_120_v91.synpkg`), the NDP processor's internal DSP firmware (`dsp_firmware_v91.synpkg`), and the ML model (`ei_model.synpkg`). 
+- The BMM150 sensor is initialized; this includes setting it into normal operation with an output data rate (ODR) of 10 Hz. 
+
+Finally, in the `loop()` function:
+
+- Memory is allocated for the sensor data; data is then read from the sensor and stored in this allocated space.
+- Raw sensor data is then extracted and parsed into raw magnetometer data. It is read 8 bits at a time from the `sensor_data` array and then combined to form a 16-bit integer for each axis (X, Y, Z) and raw Hall resistance value. 
+- Raw magnetometer data is printed on the Serial Monitor, allowing the user to observe sensor data in real-time.
+
+After uploading the example code, you should see the magnetometer data on the IDE's Serial Monitor, as shown below:
+
+![Nicla Voice onboard raw magnetometer data on the IDE's Serial Monitor](assets/user-manual-15.png)
+
+Now, open the IDE's Serial Plotter by navigating to **Tools > Serial Plotter**. After a while, you should see a real-time graph showing raw data from the board's onboard magnetometer, as shown below (move the board):

```

Comment:
```suggestion
Now open the IDE's Serial Plotter by navigating to **Tools > Serial Plotter**. After a while, you should see a real-time graph showing raw data from the board's onboard magnetometer, as shown below (move the board):
```
I suggest removing that comma, is not necessary having a simple and relative shot sentence IMHO. (typical trick: read it out loud, if you don't need to stop to breath, it is ok for the reader)